python/lsst/ip/isr/isrFunctions.py

```diff
@@ -219,8 +219,56 @@ def growMasks(mask, radius=0, maskNameList=['BAD'], maskValue="BAD"):
         fpSet.setMask(mask, maskValue)
 
 
-def maskITLEdgeBleed(ccdExposure, badAmpDict, itlEdgeBleedSatMinArea=10000,
-                     itlEdgeBleedSatMaxArea=100000,
+def getLargeSatMidPoint(fpCore):
+    """Get the number of cores and edges of saturated cores areas
+    in a saturated footprint.
+
+    Parameters
+    ----------
+    fpCore : `lsst.afw.detection._detection.Footprint`
+        Footprint of saturated core.
+
+    Returns
+    -------
+    trueSegments : int
+        Number of saturated cores in the footprint.
+    XMidPointBBox: `list` [`int`]
+        Edges of the sub footprints for each saturated core
+        if there are more than one.
+    """
+    # Get centroid of saturated core
+    xCore, yCore = fpCore.getCentroid()
+    # Turn the Y detector coordinate into Y footprint coordinate
+    yCoreFP = int(yCore) - fpCore.getBBox().getMinY()
+    # Now test if there is one or more cores
+    # by checking if the slice at the center is full of saturated pixels
+    checkCoreNbRow = fpCore.getSpans().asArray()[yCoreFP, :]
+    trueSegments = 0
+    inTrueSegment = False
+    indexSwitchTrue = []
+    indexSwitchFalse = []
+    for i, value in enumerate(checkCoreNbRow):
+        if value:
+            if not inTrueSegment:
+                indexSwitchTrue.append(i)
+                trueSegments += 1
+                inTrueSegment = True
+        elif inTrueSegment:
+            indexSwitchFalse.append(i)
+            inTrueSegment = False
+    # if there are more than one true segment in the central slice
+    # then there are several saturated core and trails in the footprint.
+    XMidPointBBox = [0]
+    # TODO: generalize to N cores.
+    if trueSegments == 2:
+        XMidPointBBox.append(int((indexSwitchTrue[1] - indexSwitchFalse[0])/2))
+        XMidPointBBox.append(fpCore.getSpans().asArray().shape[1])
```

Comment:
I think if there are more than two we should log a warning (so this should have a log passed in) so it's recorded that there are too many cores for the code right now.